pkg/appgw/frontend_listeners.go

```diff
@@ -59,6 +55,18 @@ func (c *appGwConfigBuilder) getListeners(cbCtx *ConfigBuilderContext) (*[]n.App
 	}
 
 	sort.Sort(sorter.ByListenerName(listeners))
+
```

Comment:
This is a great change.  You could probably carve this out and merge asap.
You could perhaps make this even better if you carved the `for` loop out into a function and compose functions for the systems that need `listenersByID`.

something like `listenersByID := groupByID(getListeners())`  (should you decide to do this and call the new fn `groupByID`)